lms/static/js/learner_dashboard/views/entitlement_unenroll_view.js

```diff
@@ -0,0 +1,157 @@
+(function(define) {
+    'use strict';
+    define(['backbone',
+        'jquery',
+        'gettext',
+        'edx-ui-toolkit/js/utils/html-utils'
+    ],
+        function(Backbone, $, gettext, HtmlUtils) {
+            return Backbone.View.extend({
+                el: '#unenroll-entitlement-modal',
+
+                triggerSelector: '.js-action-unenroll-entitlement',
+
+                submitButtonSelector: '#unenroll-entitlement-submit',
+
+                trackInfoSelector: '#entitlement-track-info',
+
+                refundInfoSelector: '#entitlement-refund-info',
+
+                errorInfoSelector: '#unenroll-entitlement-error',
+
+                closeButtonSelector: '#unenroll-entitlement-modal .close-entitlement-modal',
+
+                mainPageSelector: '#dashboard-main',
+
+                initialize: function(options) {
+                    var view = this;
+                    this.dashboardUrl = options.dashboardUrl;
+
+                    this.$submitButton = $(this.submitButtonSelector);
+                    this.$trackInfo = $(this.trackInfoSelector);
+                    this.$refundInfo = $(this.refundInfoSelector);
+                    this.$errorInfo = $(this.errorInfoSelector);
+
+                    this.$submitButton.on('click', this.handleSubmit.bind(this));
+
+                    $(this.triggerSelector).each(function() {
+                        var $trigger = $(this);
+
+                        $trigger.on('click', view.handleTrigger.bind(view));
+
+                        if (window.accessible_modal) {
+                            window.accessible_modal(
+                                '#' + $trigger.attr('id'),
+                                view.closeButtonSelector,
+                                '#' + view.$el.attr('id'),
+                                view.mainPageSelector
+                            );
+                        }
+                    });
+                },
+
+                handleTrigger: function(event) {
+                    var $trigger = $(event.target),
+                        courseName = $trigger.data('courseName'),
```

Comment:
This wasn't on your changes, but it's typically dangerous to pass through variables like this. A user could simply go in the DOM, change an entitlementIsRefundable to true and then pass that to the backend.

Reply:
I think it should be ok here because most of these variables, except for `apiEndpoint`, are for display/messaging purposes and will not actually be sent to the server. When we send the `DELETE` request to the server, we rely on backend logic to take care of whatever needs to happen to revoke the relevant entitlement. Refunds are processed automatically if they're eligible and do not require any data from the front end.

Reply:
Okay, makes sense. It's not a quick change either with the way all the event bindings work.